crates/sncast/src/starknet_commands/script.rs

```diff
@@ -1,38 +1,40 @@
-use std::any::Any;
 use std::collections::HashMap;
 use std::fs;
 
 use crate::starknet_commands::declare::BuildConfig;
 use crate::starknet_commands::{call, declare, deploy, invoke};
 use crate::{get_account, get_nonce, WaitForTx};
 use anyhow::{anyhow, ensure, Context, Result};
+use blockifier::execution::common_hints::ExecutionMode;
+use blockifier::execution::deprecated_syscalls::DeprecatedSyscallSelector;
+use blockifier::execution::entry_point::{
+    CallEntryPoint, EntryPointExecutionContext, ExecutionResources,
+};
+use blockifier::execution::execution_utils::ReadOnlySegments;
+use blockifier::execution::syscalls::hint_processor::SyscallHintProcessor;
+use blockifier::state::cached_state::CachedState;
 use cairo_felt::Felt252;
-use cairo_lang_casm::hints::{Hint, StarknetHint};
-use cairo_lang_casm::operand::{CellRef, ResOperand};
-use cairo_lang_runner::casm_run::{cell_ref_to_relocatable, execute_core_hint_base};
-use cairo_lang_runner::casm_run::{extract_relocatable, vm_get_range, MemBuffer};
+use cairo_lang_casm::hints::Hint;
 use cairo_lang_runner::short_string::as_cairo_short_string;
-use cairo_lang_runner::{
-    build_hints_dict, insert_value_to_cellref, RunResultValue, SierraCasmRunner,
-};
+use cairo_lang_runner::{build_hints_dict, RunResultValue, SierraCasmRunner};
 use cairo_lang_sierra::program::VersionedProgram;
 use cairo_lang_sierra_to_casm::metadata::MetadataComputationConfig;
-use cairo_lang_utils::bigint::BigIntAsHex;
 use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
-use cairo_vm::hint_processor::hint_processor_definition::{HintProcessorLogic, HintReference};
-use cairo_vm::serde::deserialize_program::ApTracking;
-use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::types::relocatable::Relocatable;
 use cairo_vm::vm::errors::hint_errors::HintError;
-use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
-use cairo_vm::vm::runners::cairo_runner::{ResourceTracker, RunResources};
 use cairo_vm::vm::vm_core::VirtualMachine;
 use camino::Utf8PathBuf;
+use cheatnet::constants::{build_block_context, build_transaction_context};
+use cheatnet::state::{CheatnetBlockInfo, DictStateReader};
```

Comment:
https://github.com/foundry-rs/starknet-foundry/issues/1419

Reply:
This will be refactored in a separate PR

Reply:
how much effort is this? perhaps this should be done as an immediate follow up to this one?

Reply:
almost no effort, so yeah, I will make a PR immediately after this one

Reply:
Cool, let's make it a stacked PR and merge those two PRs at once, sounds ok?

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1506